### PR TITLE
convert @babel/plugin-transform-typescript to typescript

### PR DESCRIPTION
**Pull request by zxbodya**
| Q                        | A 
| ------------------------ | ---
| License                  | MIT

@babel/plugin-transform-typescript part of #11578

<a href="https://gitpod.io/#https://github.com/babel/babel/pull/13222"></a>

### CODE REVIEW

File: packages/babel-plugin-transform-typescript/src/namespace.ts [hunk 1]
```diff
@@ -73,7 +73,7 @@ function handleVariableDeclaration(
     for (const declarator of node.declarations) {
       declarator.init = t.assignmentExpression(
         "=",
-        getMemberExpression(t, name, declarator.id.name),
+        getMemberExpression(t, name, (declarator.id as t.Identifier).name),
```
Comment (JLHwung) [inline packages/babel-plugin-transform-typescript/src/namespace.ts:76]: We can add `declarator is t.Identifier` assertions to the `declarator => t.isIdentifier(declarator.id)`, so TypeScript can infer that `declarator.id` here is an `t.Identifier`.

Comment (zxbodya) [reply packages/babel-plugin-transform-typescript/src/namespace.ts:76]: we can do something like this:

```ts
 if (
    declarations.every((declarator): declarator is t.VariableDeclarator & {
      id: t.Identifier;
    } => t.isIdentifier(declarator.id))
  ) {
    // ...
  }
```

what do you think, does it look better?

File: packages/babel-plugin-transform-typescript/src/index.ts [hunk 2]
```diff
@@ -184,7 +185,7 @@ export default declare((api, opts) => {
         }
 
         if (file.ast.comments) {
-          for (const comment of (file.ast.comments: Array<Object>)) {
+          for (const comment of file.ast.comments as Array<any>) {
```
Comment (JLHwung) [inline packages/babel-plugin-transform-typescript/src/index.ts:188]: Guess we can remove the array annotation for the same rationale on https://github.com/babel/babel/pull/13221#discussion_r631435037.

File: packages/babel-plugin-transform-typescript/src/namespace.ts (hunk 1 above)
Comment (JLHwung) [reply packages/babel-plugin-transform-typescript/src/namespace.ts:76]: @zxbodya Did you push the changes?

Comment (zxbodya) [reply packages/babel-plugin-transform-typescript/src/namespace.ts]: done, just pushed it also rebasing the changes to fix merge conflict

**Review by nicolo-ribaudo — APPROVED** [review]

**Review by JLHwung — APPROVED** [review]
Thanks!

**Pull request merged**